overlord/snapstate/snapmgr.go

```diff
@@ -390,20 +390,22 @@ func refreshScheduleNoWeekdays(rs []*timeutil.Schedule) error {
 	return nil
 }
 
-func (m *SnapManager) checkRefreshSchedule() ([]*timeutil.Schedule, error) {
+func (m *SnapManager) refreshSchedule() ([]*timeutil.Schedule, string, error) {
 	refreshScheduleStr := defaultRefreshSchedule
 
 	tr := config.NewTransaction(m.state)
 	err := tr.Get("core", "refresh.schedule", &refreshScheduleStr)
 	if err != nil && !config.IsNoOption(err) {
-		return nil, err
+		return nil, "", err
 	}
+
 	refreshSchedule, err := timeutil.ParseSchedule(refreshScheduleStr)
 	if err == nil {
 		err = refreshScheduleNoWeekdays(refreshSchedule)
 	}
 	if err != nil {
 		logger.Noticef("cannot use refresh.schedule configuration: %s", err)
+		refreshScheduleStr = defaultRefreshSchedule
```

Comment:
Just a comment when I realized what this is doing. I would not have expected this to change anything, just compute and return the value. Perhaps we want a separate setter that gets to do this part.

Reply:
Thanks, you are right, as a short term fix I renamed the function to make the nature of it more obvious. The reason why it has this side-effect is that it is called in two places and in both places invalid values must be reset to the default schedule. I plan a bigger refactor (in the spirit of #4266) for auto-refresh, so maybe that is a good time to tackle this one.

Reply:
we should also remember that we are moving configure of core/snap inside snapd, and part of that plan (to fix current issues) afaiu is to always run configure snapd at least once per restart,  in that world we could be able to make sure that the value in configuration and the value on SnapManager are  in sync in a non-lazy way I think